### PR TITLE
feat: add hover support to web

### PR DESCRIPTION
**Pull request by RichardLindhout**
### Summary
Fixes https://github.com/callstack/react-native-paper/issues/3749

### Test plan

I tested the demo app!

https://user-images.githubusercontent.com/6492229/224786455-0c9665c0-3d26-424c-b9c5-56ed31abc474.mov

### CODE REVIEW

File: src/components/TouchableRipple/TouchableRipple.tsx [hunk 1]
```diff
@@ -236,9 +244,20 @@ const TouchableRipple = ({
...
+        state.hovered && { backgroundColor: hoverColor },
+        typeof style === 'function' ? style(state) : style,
+      ]}
     >
-      {React.Children.only(children)}
+      {(state) =>
+        React.Children.only(
+          typeof children === 'function' ? children(state) : children
+        )
+      }
```
Comment (lukewalczak) [inline src/components/TouchableRipple/TouchableRipple.tsx:260]: Please add unit test for introduced changes in styles and rendering the children.

File: src/components/TouchableRipple/TouchableRipple.tsx [hunk 2]
```diff
@@ -2,17 +2,20 @@ import * as React from 'react';
 import {
   GestureResponderEvent,
   Platform,
-  Pressable,
   StyleProp,
   StyleSheet,
   ViewStyle,
 } from 'react-native';
 
+import Color from 'color';
```
Comment (lukewalczak) [inline src/components/TouchableRipple/TouchableRipple.tsx:10]:
```suggestion
import color from 'color';
```

File: src/components/TouchableRipple/TouchableRipple.tsx [hunk 3]
```diff
@@ -100,16 +108,16 @@ const TouchableRipple = ({
   ...rest
 }: Props) => {
   const theme = useInternalTheme(themeOverrides);
+  const { calculatedRippleColor } = getTouchableRippleColors({
+    theme,
+    rippleColor,
+  });
+  const hoverColor = Color(calculatedRippleColor).fade(0.5).rgb().string();
```
Comment (lukewalczak) [inline src/components/TouchableRipple/TouchableRipple.tsx:115]:
```suggestion
  const hoverColor = color(calculatedRippleColor).fade(0.5).rgb().string();
```

**Comment by arjendevos** [comment]
`TouchableRipple.native` does still need some fixes to work properly with functions. This will be fixed soon!

**Pull request closed without merging**